### PR TITLE
Add least-square solution to wilson factorization

### PR DESCRIPTION
**Pull request by tensionhead**
Author Guidelines
-----------------
- [x] Is the change set **< 400 lines**?

Reviewer Checklist
------------------
- [ ] Are testing routines present?
- [ ] Do **parallel** loops have a set length and correct termination conditions?
- [ ] Do objects in the global package namespace perform proper parsing of their input? 
- [ ] Do code-blocks provide novel functionality, i.e., no re-factoring using builtin/external packages possible?
- [ ] Code layout
  - [ ] Is the code PEP8 compliant?
  - [ ] Does the code adhere to agreed-upon naming conventions?
  - [ ] Are keywords clearly named and easy to understand?
  - [ ] No commented-out code?
- [ ] Are all docstrings complete and accurate?

### CODE REVIEW

File: syncopy/nwanalysis/wilson_sf.py [hunk 1]
```diff
@@ -59,17 +63,25 @@ def wilson_sf(CSD, nIter=100, rtol=1e-9):
...
 
-        psi_inv = np.linalg.inv(psi)
-        # the bracket of equation 3.1
-        g = psi_inv @ CSD @ psi_inv.conj().transpose(0, 2, 1)
+        if direct_inversion:
+            psi_inv = np.linalg.inv(psi)
+            # the bracket of equation 3.1
+            g = psi_inv @ CSD @ psi_inv.conj().transpose(0, 2, 1)
+        else:
+            for i in range(nFreq):
```
Comment (pantaray) [inline syncopy/nwanalysis/wilson_sf.py:75]: I think we don't need the for loop: `numpy.linalg.lstsq` can process right hand sides `b` with shape `(M,M,K)`, right? One transpose of `CSD` should do it and we don't have to iterate over frequencies.

**Review by pantaray — COMMENTED** [review]
Looks all good - I think we could get rid of the for loop in `wilson_sf` which should improve performance.

File: syncopy/nwanalysis/wilson_sf.py (hunk 1 above)
Comment (tensionhead) [reply syncopy/nwanalysis/wilson_sf.py:75]: No they actually can not, shapes are: 
```
a : (M, N) array_like
    "Coefficient" matrix.
b : {(M,), (M, K)} array_like
```

Comment (pantaray) [reply syncopy/nwanalysis/wilson_sf.py:75]: I see, my bad - the coefficient matrix hast to be 2d, the rhs can be 3d. Never mind then, thanks for the clarification.

**Review by pantaray — APPROVED** [review]
Looks all good.

**Pull request merged**